spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/repository/RdbmsDeploymentIdRepository.java

```diff
@@ -60,7 +61,7 @@ public void delete(String name) {
 	}
 
 	@Override
-	public String findOne(String key) {
-		return findById(key).orElse(null);
+	public Optional<String> findByKey(String key) {
+		return findById(key);
 	}
```

Comment:
`findById` instead?

Reply:
This is a weird interface as it really searches by key(aka value) instead of id.